### ISSUE TITLE
Datagrid/Accessibility: Expand/Collapse buttons on groups are not keyboard-accessible

### ISSUE THREAD

**Issue by EdwardCoyle** [infor-design/enterprise#2164]
**Describe the bug**
When working with a Datagrid with row groups, it's not possible to use keyboard navigation to focus the buttons that expand/collapse the rows.

Additionally, if a cell is clicked, and arrow keys are used to navigate to a grouping header row, if that header row is activated, keyboard navigation within the datagrid boundaries is no longer possible.

**To Reproduce**
_Test initial attempts to focus the button:_
1. Goto http://localhost:4000/components/datagrid/example-grouping-filter
1. Focus the filter input in the "Purchases" column header
1. Press the tab key
1. Notice that focus is not on the button in the grouping row header (and possibly no longer in the datagrid boundaries)

_Test post-activation keyboard nav:_
1. Goto http://localhost:4000/components/datagrid/example-grouping-filter
1. Click on any cell in the datagrid
1. Use the arrow keys to navigate to any row grouping header
1. Press the spacebar to activate (expand/collapse)
1. Use the arrow keys to try and navigate.  Notice that it's no longer possible.

**Expected behavior**
It should be possible to tab in/out of these buttons, and keyboard navigation should remain active while inside the grid.

**Version**
Tested on `v4.19.x`

**Screenshots**
this is a video example of the first issue:
![May-15-2019-12_26-PM](https://user-images.githubusercontent.com/3249601/57793245-da772680-770e-11e9-843c-b3c8c2d8d85a.gif)

**Platform**
-n/a

**Additional context**
Discovered while testing #2151.  I'm wondering if the first problem (initial focus of the header row) has to do with the fact that the example I was looking at has a grouping header as its first row?  I'm also not 100% sure if it's the entire header that should be active, or just the button, but from the looks of the changes that fix #2005, it appears the button itself should be focusable.

**Comment by tmcconechy** [comment]
Descoping for now
